src/PrestaShopBundle/DataCollector/HookRegistry.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * 2007-2017 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\DataCollector;
+
+use PrestaShop\PrestaShop\Core\Module\ModuleInterface;
+
+/**
+ * Store every hook dispatched during a request.
+ */
+class HookRegistry
+{
+    /**
+     * @var array the selected hook.
+     */
+    private $hook;
+
+    /**
+     * @var array the list of hooks.
+     */
+    private $hooks = array(
+        'called' => array(),
+        'notCalled' => array(),
+    );
+
+    /**
+     * Notify the registry that the selected hook have been called.
+     */
+    public function hookWasCalled()
+    {
+        $this->hook['called'] = 'called';
```

Comment:
I think `true` instead of `'called'` would be better, or is there a reason for the text?

Reply:
yes, it's easier to collect hooks as this value is the "property" already stored in `$hooks` so I don't need an additional check on `collect` function.